File: EPSILON/TOOLS/Altova/Altova/XMLSpy2006/spl/cs/AltovaText/EDIFact/ServiceStringAdvice.cs
[call GenerateFileHeader("ServiceStringAdvice.cs")]
using System.IO;

namespace Altova.TextParser.EDIFACT
{
	/// <summary>
	/// Encapsulates a service string advice (UNA) segment.
	/// </summary>
	public class ServiceStringAdvice
	{
		#region Implementation Detail:
		char mComponentSeparator = ':';
		char mDataElementSeparator = '+';
		char mSegmentTerminator = '\\'';
		char mReleaseCharacter = '?';
		char mDecimalSeparator = '.';
		#endregion
		#region Public Interface:
		/// <summary>
		/// Get/sets the composite data element separator.
		/// </summary>
		public char ComponentSeparator
		{
			get
			{
				return mComponentSeparator;
			}
			set
			{
				mComponentSeparator = value;
			}
		}
		/// <summary>
		/// Get/sets the data element separator.
		/// </summary>
		public char DataElementSeparator
		{
			get
			{
				return mDataElementSeparator;
			}
			set
			{
				mDataElementSeparator = value;
			}
		}
		/// <summary>
		/// Get/sets the segment terminator.
		/// </summary>
		public char SegmentTerminator
		{
			get
			{
				return mSegmentTerminator;
			}
			set
			{
				mSegmentTerminator = value;
			}
		}
		/// <summary>
		/// Get/sets the release character.
		/// </summary>
		public char ReleaseCharacter
		{
			get
			{
				return mReleaseCharacter;
			}
			set
			{
				mReleaseCharacter = value;
			}
		}
		/// <summary>
		/// Get/sets the decimal separator.
		/// </summary>
		public char DecimalSeparator
		{
			get
			{
				return mDecimalSeparator;
			}
			set
			{
				mDecimalSeparator = value;
			}
		}
		/// <summary>
		/// Sets all members from a character array.
		/// </summary>
		/// <param name="rhs">the character array to set from</param>
		public void SetFromCharArray(char\[\] rhs)
		{
			mComponentSeparator = rhs\[0\];
			mDataElementSeparator = rhs\[1\];
			mDecimalSeparator = rhs\[2\];
			mReleaseCharacter = rhs\[3\];
			mSegmentTerminator = rhs\[5\];
		}
		/// <summary>
		/// Serializes this instance in EDIFact format to a stream.
		/// </summary>
		/// <param name="stream">the stream to serialize to</param>
		public void Serialize(TextWriter stream)
		{
			stream.Write("UNA");
			stream.Write(mComponentSeparator);
			stream.Write(mDataElementSeparator);
			stream.Write(mDecimalSeparator);
			stream.Write(mReleaseCharacter);
			stream.Write(' ');
			stream.Write(mSegmentTerminator);
		}
		#endregion
	}
}

File: EPSILON/SAMPLE FILE/E-publishing/Tools/Altova/Altova/XMLSpy2006/spl/java/altova/text/edifact/ServiceStringAdvice.java
[call GenerateFileHeader("ServiceStringAdvice.java")]
package com.altova.text.edifact;

import java.io.IOException;
import java.io.Writer;

public class ServiceStringAdvice {
	private char m_ComponentSeparator = ':';

	private char m_DataElementSeparator = '+';

	private char m_SegmentTerminator = '\\'';

	private char m_ReleaseCharacter = '?';

	private char m_DecimalSeparator = '.';

	public char getComponentSeparator() {
		return m_ComponentSeparator;
	}

	public void setComponentSeparator(char componentSeparator) {
		m_ComponentSeparator = componentSeparator;
	}

	public char getDataElementSeparator() {
		return m_DataElementSeparator;
	}

	public void setDataElementSeparator(char dataElementSeparator) {
		m_DataElementSeparator = dataElementSeparator;
	}

	public char getDecimalSeparator() {
		return m_DecimalSeparator;
	}

	public void setDecimalSeparator(char decimalSeparator) {
		m_DecimalSeparator = decimalSeparator;
	}

	public char getReleaseCharacter() {
		return m_ReleaseCharacter;
	}

	public void setReleaseCharacter(char releaseCharacter) {
		m_ReleaseCharacter = releaseCharacter;
	}

	public char getSegmentTerminator() {
		return m_SegmentTerminator;
	}

	public void setSegmentTerminator(char segmentTerminator) {
		m_SegmentTerminator = segmentTerminator;
	}

	public void serialize(Writer stream) throws IOException {
		stream.write("UNA");
		stream.write(m_ComponentSeparator);
		stream.write(m_DataElementSeparator);
		stream.write(m_DecimalSeparator);
		stream.write(m_ReleaseCharacter);
		stream.write(' ');
		stream.write(m_SegmentTerminator);
	}

	public void setFromCharArray(char\[\] rhs) {
		m_ComponentSeparator = rhs\[0\];
		m_DataElementSeparator = rhs\[1\];
		m_DecimalSeparator = rhs\[2\];
		m_ReleaseCharacter = rhs\[3\];
		m_SegmentTerminator = rhs\[5\];
	}
}
